akka-discovery/src/main/scala/akka/discovery/ServiceDiscovery.scala

```diff
@@ -88,6 +89,9 @@ object ServiceDiscovery {
 @ApiMayChange
 final case class Lookup(serviceName: String, portName: Option[String], protocol: Option[String]) {
 
+  require(serviceName != null, "serviceName cannot be null")
```

Comment:
'serviceName' can not be null, I think nice to quote the param to be helpful to users.

Reply:
Should this go into an apply? can java users use it?

Reply:
This is in the constructor body and Java devs are expected to use `new Lookup(...)`. The `apply` is calling that constructor.